amqtt/contrib/http.py

```diff
@@ -0,0 +1,184 @@
+from amqtt.session import Session
+
+logger = logging.getLogger(__name__)
+
+
+class ResponseMode(StrEnum):
+    STATUS = "status"
+    JSON = "json"
+    TEXT = "text"
+
+class RequestMethod(StrEnum):
+    GET = "get"
+    POST = "post"
+    PUT = "put"
+
+
+class ParamsMode(StrEnum):
+    JSON = "json"
+    FORM = "form"
+
+
+class ACLError(Exception):
+    pass
+
+
+HTTP_2xx_MIN = 200
+HTTP_2xx_MAX = 300
+
+HTTP_4xx_MIN = 400
+HTTP_4xx_MAX = 500
+
+
+class HttpAuthTopicPlugin(BaseAuthPlugin, BaseTopicPlugin):
+
+    def __init__(self, context: BrokerContext) -> None:
+        super().__init__(context)
+        self.http = ClientSession(headers = {"User-Agent": self.config.user_agent})
+
+        match self.config.request_method:
+            case RequestMethod.GET:
+                self.method = self.http.get
+            case RequestMethod.PUT:
+                self.method = self.http.put
+            case _:
+                self.method = self.http.post
+
+    async def on_broker_pre_shutdown(self) -> None:
+        await self.http.close()
+
+    @staticmethod
+    def _is_2xx(r: ClientResponse) -> bool:
+        return HTTP_2xx_MIN <= r.status < HTTP_2xx_MAX
+
+    @staticmethod
+    def _is_4xx(r: ClientResponse) -> bool:
+        return HTTP_4xx_MIN <= r.status < HTTP_4xx_MAX
+
+    def _get_params(self, payload: dict[str, Any]) -> dict[str, Any]:
+        match self.config.params_mode:
+            case ParamsMode.FORM:
+                match self.config.request_method:
+                    case RequestMethod.GET:
+                        kwargs = { "params": payload }
+                    case _: # POST, PUT
+                        d: Any = FormData(payload)
+                        kwargs = {"data": d}
+            case _:  # JSON
+                kwargs = { "json": payload}
+        return kwargs
+
+    async def _send_request(self, url: str, payload: dict[str, Any]) -> bool|None: # pylint: disable=R0911
+
+        kwargs = self._get_params(payload)
+
+        async with self.method(url, **kwargs) as r:
+            logger.debug(f"http request returned {r.status}")
+
+            match self.config.response_mode:
+                case ResponseMode.TEXT:
+                    return self._is_2xx(r) and (await r.text()).lower() == "ok"
+                case ResponseMode.STATUS:
+                    if self._is_2xx(r):
+                        return True
+                    if self._is_4xx(r):
+                        return False
+                    # any other code
+                    return None
+                case _:
+                    if not self._is_2xx(r):
+                        return False
+                    data = await r.json()
+                    for ok in ("OK", "Ok", "ok"):
+                        if ok in data and data[ok] is None:
+                            return None
+                        if ok in data and isinstance(data[ok], bool):
+                            return bool(data[ok])
```

Comment:
If I'm not mistaken this should have identical behaviour and appears to me a bit more concise. What do you think?

```suggestion
                    data = {k.lower(): v for (k, v) in data.items()}
                    return data.get("ok", default=None)
```